prediction_market_agent/agents/replicate_to_omen_agent/deploy.py

```
@@ -90,21 +98,40 @@ def replicate(self, settings: ReplicateSettings) -> None:
+                collateral_token_address = Web3.to_checksum_address(
+                    replicate_config.collateral_token
+                )
+                # make sure it's ERC20
+                try:
+                    ContractERC20OnGnosisChain(
+                        address=collateral_token_address
+                    ).symbol()
+                except Exception as e:
+                    raise ValueError(
+                        f"Collateral token {collateral_token_address} is not an ERC20. {e=}"
+                    )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)